discord_bot/schedule/channels/membership.py

```diff
@@ -44,20 +33,24 @@ async def sync_channel_membership(self, game_channel: GameChannel):
 
         await async_set_default_channel_membership(game_channel)
         expected_members = await async_get_game_channel_members(game_channel)
-        expected_member_ids = self.get_discord_ids(expected_members)
+        expected_member_ids = set(map(lambda x: x.user.discord_id, expected_members))
         actual_members = await async_get_channel_current_members(discord_channel)
-        actual_member_ids = self.get_discord_ids(actual_members)
+        actual_member_ids = set(map(lambda x: str(x.id), actual_members))
+        missing_users = list(filter(lambda m: m.user.discord_id != None and not actual_member_ids.__contains__(m.user.discord_id), expected_members))
+        missing_user_ids = list(map(lambda m: m.user.id, missing_users))
```

Comment:
I think we risk losing some readability with lambda functions, I tend to shy away from them for that reason